py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class Accumulator(ABC):
+    """
+    The standard Accumulator template.
+    Define these functions to build a custom
+    Accumulator for aggregation.
+    """
+
+    @abstractmethod
+    def create(self) -> Any:
+        raise NotImplementedError
+
+    @abstractmethod
+    def add(self, acc: Any, value: Any) -> Any:
```

Comment:
I would try to type this out properly soon. from what I understand `value` comes from the previous step, and `acc` can be a custom type specific to the reduce step.

if the two types are equal (like in your word counter example), maybe we can allow the user to only specify one function, since `add` and `merge` would do the same thing?